runtime/mixed.inl

```diff
@@ -551,6 +598,9 @@ int64_t mixed::to_int() const {
     case type::ARRAY:
       php_warning("Wrong conversion from array to int");
       return as_array().to_int();
+    case type::OBJECT:
+      php_warning("Wrong conversion from %s to int", get_type_or_class_name());
+      return (as_object() ? 1 : 0);
```

Comment:
Shouldn't we just return `1` as it's specified for `intval`?
 https://www.php.net/manual/en/function.intval.php

Reply:
In case of null object, `intval` returns 0

Reply:
This is what written in `intval`'s docs:
`intval() should not be used on objects, as doing so will emit an [E_WARNING](https://www.php.net/manual/en/errorfunc.constants.php#constant.e-warning) level error and return 1.`